src/gsCore/gsComposedBasis.h

```diff
@@ -0,0 +1,307 @@
+/** @file gsComposedBasis.h
+
+    @brief
+
+    This file is part of the G+Smo library.
+
+    This Source Code Form is subject to the terms of the Mozilla Public
+    License, v. 2.0. If a copy of the MPL was not distributed with this
+    file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+    Author(s):
+        H.M. Verhelst   (2019-..., TU Delft)
+        A. Mantzaflaris (2019-..., Inria)
+*/
+
+#pragma once
+
+#include <gsCore/gsBasis.h>
+#include <gsCore/gsComposedGeometry.h>
+
+namespace gismo
+{
+
+template <class T>
+class gsComposedBasis : public gsBasis<T>
+{
+
+    typedef memory::shared_ptr< gsComposedBasis > Ptr;
+    typedef memory::unique_ptr< gsComposedBasis > uPtr;
+
+    // /// Geometry Type
+    typedef gsComposedGeometry<T> GeometryType;
+
+    GISMO_CLONE_FUNCTION(gsComposedBasis)
+//
+    GISMO_MAKE_GEOMETRY_NEW
+
+    typedef memory::unique_ptr< gsDomainIterator<T> > domainIter;
+
+public:
+    typedef gsBasis<T>      BasisT;
+    typedef gsFunction<T>   CompositionT;
+
+public:
+    gsComposedBasis(const CompositionT & composition, const BasisT & basis)
+    :
+    m_composition(&composition),
+    m_basis(&basis)
+    {
+        GISMO_ENSURE(m_basis->domainDim()==m_composition->targetDim(),
+            "Domain dimension of the basis "<<
+            " should be equal to the target dimension of the composition "<<
+            ", but basis.domainDim() = "<<basis.domainDim()<<
+            " and composition.targetDim() = )"<<composition.targetDim());
+    }
+
+    short_t domainDim() const override { return m_composition->domainDim(); }
+    short_t targetDim() const override { return m_basis->targetDim(); }
+
+    short_t maxDegree() const override { return m_basis->maxDegree(); }
+
+    gsMatrix<T> support() const override
+    {
+        gsMatrix<T> supp = m_basis->support();
+        // gsGridIterator<T,CUBE> pt(supp,math::pow(2,this->domainDim()));
```

Comment:
We might need to decide on this code: is it needed? Then uncomment. Otherwise, remove